index.js

```diff
@@ -77,23 +79,32 @@
       });
     }
 
-    function shouldStop(report) {
+    function shouldFail(report) {
       if(report.warningCount &&
-        (options.showWarnings || options.stopOnWarning)) {
+        (options.showWarnings || options.failOnWarning)) {
           processWarnings(report.results);
       }
-      if(report.errorCount) processErrors(report.results);
-      return (report.errorCount && options.stopOnError) ||
-        (report.warningCount && options.stopOnWarning);
+
+      if(report.errorCount) {
+        totalErrorCount += report.errorCount;
+        processErrors(report.results);
+      };
+      return (report.errorCount && options.failOnError) ||
+        (report.warningCount && options.failOnWarning);
     }
 
     return function(content, file, done) {
       var report = cli.executeOnFiles([file.path]);
 
       log.debug('Processing "%s".', file.originalPath);
-      if(shouldStop(report)) {
+      if(shouldFail(report)) {
         done(report.results);
-      } else {
+      } else if (options.stopAboveErrorThreshold && totalErrorCount > options.errorThreshold) {
+        log.error('\n' + chalk.red('There are more than ' + options.errorThreshold + ' errors'));
+        process.exit();
+      }
+      else {
+        log.error('\n' + chalk.red('There are ' + totalErrorCount + ' errors'));
```

Comment:
Could display warning count. I'm not sure they need to be optional - it's only output once so doesn't add a lot of noise.